cylc/flow/etc/syntax/cylc-mode.el

```diff
@@ -114,11 +114,7 @@
   (run-hooks 'cylc-mode-hook))
 
 ;;;###autoload
-(add-to-list 'auto-mode-alist '("suite*.rc" . cylc-mode))
-(setq auto-mode-alist
-  (append
-    '(("suite*.rc" . cylc-mode)
-      ("*.cylc" . cylc-mode))
-   auto-mode-alist))
+(add-to-list 'auto-mode-alist '("suite.*\\.rc\\'" . cylc-mode))
+(add-to-list 'auto-mode-alist '("\\.cylc\\'" . cylc-mode))
```

Comment:
This matches any filename ending ".rc" which contains "suite in the name" plus any filename ending ".cylc".
See https://www.emacswiki.org/emacs/AutoModeAlist